packages/flutter_tools/lib/src/asset.dart

```diff
@@ -268,6 +270,16 @@ class ManifestAssetBundle implements AssetBundle {
     final DevFSStringContent licenses = DevFSStringContent(licenseResult.combinedLicenses);
     additionalDependencies = licenseResult.dependencies;
 
+    if (warnOnWildcard && _wildcardDirectories.isNotEmpty) {
+      globals.printError(
+        'WARNING: wildcard asset directories are incompatible with incremental builds.\n'
```

Comment:
Perhaps I could _only_ warn on a release build, and change the phrasing to "before building a final APK for upload to app stores?"